src/hooks/useAsyncHandler.ts

```diff
@@ -27,23 +27,26 @@ const useSafeDispatch = (dispatch: Function) => {
     );
 };
 
-export interface UseAsyncResponse<T, U> {
+export interface Response<T, U> {
     data: U extends undefined ? T | null : T;
     error: any;
     isError: boolean;
     isIdle: boolean;
     isLoading: boolean;
     isSuccess: boolean;
     reset: () => any;
-    run: (promise: Promise<any>, dataFetcher?: Function | undefined) => any;
+    run: (
+        promise: Promise<any> | PromiseLike<any>,
+        dataFetcher?: Function | undefined
+    ) => any;
     setData: (setDataResponse: T) => void;
     setError: (setErrorResponse: any) => any;
     status: States;
 }
 
-function useAsync<T>(): UseAsyncResponse<T, undefined>;
-function useAsync<T>(initialData: T): UseAsyncResponse<T, T>;
-function useAsync<T>(initialData?: T): UseAsyncResponse<T, undefined> {
+function useAsyncHandler<T>(): Response<T, undefined>;
+function useAsyncHandler<T>(initialData: T): Response<T, T>;
+function useAsyncHandler<T>(initialData?: T): Response<T, undefined> {
```

Comment:
Renamed because the library `reactUse` has a component called `useAsync` and the auto import kept choosing that one.